src/utils/components/DiskModal/utils/types.ts

```diff
@@ -79,3 +85,25 @@ export type DiskModalProps = {
   onUploadedDataVolume?: (dataVolume: V1beta1DataVolume) => void;
   vm: V1VirtualMachine;
 };
+
+export type V1DiskModalProps = {
+  createDiskSource?: SourceTypes;
+  editDiskName?: string;
+  isCreated?: boolean;
+  isOpen: boolean;
+  onClose: () => void;
+  onSubmit: (updatedVM: V1VirtualMachine) => Promise<V1VirtualMachine | void>;
+  onUploadedDataVolume?: (dataVolume: V1beta1DataVolume) => void;
+  vm: V1VirtualMachine;
+};
+
+export type V1DiskFormState = {
+  dataVolumeTemplate?: V1DataVolumeTemplateSpec;
+  disk: V1Disk;
+  isBootSource: boolean;
+  namespace: string;
```

Comment:
is namespace necessary? we should create the disk inside the same namespace the VM is at in all cases, unless it's not the disk namespace which in that case the name should be clearer